libraries/AP_DDS/AP_DDS_ExternalControl.cpp

```diff
@@ -3,9 +3,45 @@
 #include "AP_DDS_ExternalControl.h"
 #include "AP_DDS_Frames.h"
 #include <AP_AHRS/AP_AHRS.h>
+#include <AP_Common/Location.h>
 
 #include <AP_ExternalControl/AP_ExternalControl.h>
 
+// These are the Goal Interface constants. Because microxrceddsgen does not expose
+// them in the generated code, they are manually maintained
+static constexpr uint16_t TYPE_MASK_IGNORE_LATITUDE = 1; // Position ignore flags
+// uint16 IGNORE_LONGITUDE = 2
+// uint16 IGNORE_ALTITUDE = 4
+// uint16 IGNORE_VX = 8  # Velocity vector ignore flags
+// uint16 IGNORE_VY = 16
+// uint16 IGNORE_VZ = 32
+// uint16 IGNORE_AFX = 64        # Acceleration/Force vector ignore flags
+// uint16 IGNORE_AFY = 128
+// uint16 IGNORE_AFZ = 256
+// uint16 FORCE = 512    # Force in af vector flag
+// uint16 IGNORE_YAW = 1024
+// uint16 IGNORE_YAW_RATE = 2048
+
+bool AP_DDS_External_Control::handle_global_position_control(ardupilot_msgs_msg_GlobalPosition& cmd_pos)
+{
+    auto *external_control = AP::externalcontrol();
+    if (external_control == nullptr) {
+        return false;
+    }
+
+    if (strcmp(cmd_pos.header.frame_id, MAP_FRAME) == 0) {
+        // Narrow the altitude
+        const int32_t alt_cm  = static_cast<int32_t>(cmd_pos.altitude * 1E3);
+
+        // TODO for now, assume map is where home is.
```

Comment:
Equivalent mavlink handling of frames for this command here: https://github.com/ArduPilot/ardupilot/blob/00a65632eb593ab0e38358adf0f6b8200c72fd01/ArduPlane/GCS_Mavlink.cpp#L1366-L1382

Reply:
I added a converter utility that continues to use the AltFrame enum in the Location constructor rather than twiddling with the flags directly in the location struct.